cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+                vm,
+                self.migration_request_details_list.get_progress_description(vm),
+            )
+            clock_reading2 = self.migration_request_details_list.get_clock(vm)
+            logger.info("clock_reading1: %s, clock_reading2:%s", clock_reading1, clock_reading2)
+            logger.info(
+                "For vm %s, is currently in progress: %s",
+                vm,
+                self.migration_request_details_list.is_in_progress(vm),
+            )
+            migration_plan_in_progress_tracker.append(
+                self.migration_request_details_list.is_in_progress(vm)
+                and (clock_reading1 < clock_reading2)
+            )
+        return not any(migration_plan_in_progress_tracker)
+
+
+@attr.s
+class MigrationPlan(BaseEntity):
+    """Class representing v2v Migration Plan"""
+
+    name = attr.ib()
+    infra_map = attr.ib()
+    vm_list = attr.ib()
+    description = attr.ib(default=None)
+    csv_import = attr.ib(default=False)
+    target_provider = attr.ib(default=None)
+    osp_security_group = attr.ib(default=None)
+    osp_flavor = attr.ib(default=None)
+    pre_playbook = attr.ib(default=None)
+    post_playbook = attr.ib(default=None)
+
+    def is_plan_in_progress(self):
+        """MIQ V2V UI is going through redesign as OSP will be integrated."""
+        view = navigate_to(self, "InProgress")
+
+        def _in_progress():
+            try:
+                is_plan_visible = view.progress_card.is_plan_visible(self.name)
+                if is_plan_visible:
+                    try:
+                        plan_time_elapsed = view.progress_card.get_clock(self.name)
+                        new_msg = "time elapsed for migration: {time}".format(
+                            time=plan_time_elapsed
+                        )
+                    except NoSuchElementException:
+                        new_msg = "playbook is executing.."
+                        pass
+                    logger.info(
+                        "For plan {plan_name}, is plan in progress: {visibility}, {message}".format(
+                            plan_name=self.name, visibility=is_plan_visible, message=new_msg
+                        )
+                    )
+                return not is_plan_visible
+            except ItemNotFound:
+                return True
+
+        return wait_for(
+            func=_in_progress,
+            message="migration plan is in progress, be patient please",
+            delay=5,
+            num_sec=1800,
+        )
+
+    def is_migration_complete(self):
+        """Uses search box to find migration plan, return True if found.
+        Args:
+            migration_plan name: (object) Migration Plan name
+        """
+        view = navigate_to(self, "Complete")
+        view.wait_displayed()
+        view.items_on_page.item_select("15")
+        view.search_box.fill("{}\n\n".format(self.name))
+        return (self.name in view.plans_completed_list.read() and
```

Comment:
So this is actually returning whether the plan is complete _and_ successful, not just if its complete. These should likely be two separate properties, or the method name should reflect the state it is checking.